test/e2e/node/kubelet.go

```diff
@@ -310,52 +302,73 @@ var _ = SIGDescribe("kubelet", func() {
-				framework.ExpectNoError(err)
-				if resourceMonitor != nil {
-					resourceMonitor.LogLatest()
-				}
+		ginkgo.It("kubelet should be able to delete %d pods per node in %v", func() {
+			totalPods := 10 * numNodes
+			ginkgo.By(fmt.Sprintf("Creating a RC of %d pods and wait until all pods of this RC are running", totalPods))
+			rcName := fmt.Sprintf("cleanup%d-%s", totalPods, string(uuid.NewUUID()))
+
+			err := framework.RunRC(testutils.RCConfig{
+				Client:       f.ClientSet,
+				Name:         rcName,
+				Namespace:    f.Namespace.Name,
+				Image:        imageutils.GetPauseImageName(),
+				Replicas:     totalPods,
+				NodeSelector: nodeLabels,
+			})
+			framework.ExpectNoError(err)
+			// Perform a sanity check so that we know all desired pods are
+			// running on the nodes according to kubelet. The timeout is set to
+			// only 30 seconds here because framework.RunRC already waited for all pods to
+			// transition to the running status.
+			err = waitTillNPodsRunningOnNodes(f.ClientSet, nodeNames, rcName, ns, totalPods, time.Second*30)
+			framework.ExpectNoError(err)
+			if resourceMonitor != nil {
+				resourceMonitor.LogLatest()
+			}
 
-				ginkgo.By("Deleting the RC")
-				framework.DeleteRCAndWaitForGC(f.ClientSet, f.Namespace.Name, rcName)
-				// Check that the pods really are gone by querying /runningpods on the
-				// node. The /runningpods handler checks the container runtime (or its
-				// cache) and  returns a list of running pods. Some possible causes of
-				// failures are:
-				//   - kubelet deadlock
-				//   - a bug in graceful termination (if it is enabled)
-				//   - docker slow to delete pods (or resource problems causing slowness)
-				start := time.Now()
-				err = waitTillNPodsRunningOnNodes(f.ClientSet, nodeNames, rcName, ns, 0, itArg.timeout)
-				framework.ExpectNoError(err)
-				e2elog.Logf("Deleting %d pods on %d nodes completed in %v after the RC was deleted", totalPods, len(nodeNames),
-					time.Since(start))
-				if resourceMonitor != nil {
-					resourceMonitor.LogCPUSummary()
-				}
+			ginkgo.By("Deleting the RC")
+			start := time.Now()
+			framework.DeleteRCAndWaitForGC(f.ClientSet, f.Namespace.Name, rcName)
+			// Check that the pods really are gone by querying /runningpods on the
+			// node. The /runningpods handler checks the container runtime (or its
+			// cache) and  returns a list of running pods. Some possible causes of
+			// failures are:
+			//   - kubelet deadlock
+			//   - a bug in graceful termination (if it is enabled)
+			//   - docker slow to delete pods (or resource problems causing slowness)
+			e2elog.Logf("Deleting %d pods on %d nodes completed in %v after the RC was deleted", totalPods, len(nodeNames), time.Since(start))
+			if resourceMonitor != nil {
+				resourceMonitor.LogCPUSummary()
+			}
+		})
+
+		ginkgo.It("should let Kubelet delete 10 pods per node with in 1 minute", func() {
```

Comment:
I'm confused about the purpose of this test - it seems like its similar to the test above, but with a less restrictive timeout. Can you share more about the value this test provides?

Reply:
The main difference between the tests is the Kubelet sanity check which [was requested](https://github.com/kubernetes/kubernetes/pull/80731#issuecomment-524410321).

Reply:
@mattjmcnaughton the point of this PR is to duplicate the behavior of the "kubelet should be able to delete %d pods per node in %v", only including changes required to meet Conformance standards. The latest commit should make this more clear: https://github.com/kubernetes/kubernetes/pull/82250/commits/02a17e87bfd9bb15f7a86d8cae1154378c49c602

Reply:
Gotcha - thanks! I will let @Random-Liu take it from here, as he has the best understanding of what he's hoping this test will include :)